spot_utils/perception/capture.py

```diff
@@ -0,0 +1,112 @@
+"""Functions for collecting images from different cameras and camera sources on
+the spot."""
+
+from typing import Any
+
+import cv2
+import numpy as np
+from bosdyn.api.image_pb2 import Image
+from bosdyn.client.frame_helpers import BODY_FRAME_NAME, get_a_tform_b
+from bosdyn.client.image import _depth_image_data_to_numpy, build_image_request
+
+from spot_utils.structures.image import DepthImage, Intrinsics, RGBDImage, RGBImage
+from spot_utils.structures.robot import RobotClient
+
+CAMERAS = ["frontleft", "frontright", "left", "right", "back", "hand"]
+
+SOURCES_DEPTH_FRAME = {
+    cam: (cam + "_depth", cam + "_visual_in_depth_frame") for cam in CAMERAS[:-1]
+}
+SOURCES_DEPTH_FRAME["hand"] = ("hand_depth", "hand_color_in_hand_depth_frame")
+
+SOURCES_RGB_FRAME = {
+    cam: (cam + "_depth_in_visual_frame", cam + "_fisheye_image")
+    for cam in CAMERAS[:-1]
+}
+SOURCES_RGB_FRAME["hand"] = ("hand_depth_in_hand_color_frame", "hand_color_image")
+
+
+def get_intrinsics(response: Any):
```

Comment:
i'm not sure why mypy isn't complaining, but let's add output types here and in the functions below

Reply:
The typing stuff is going to get really tricky since BD uses google protobuf which doesn't play nice with the type checker.

Reply:
okay got it. if you want to add comments to indicate where the `Any` is a placeholder for a protobuf thing, that could be helpful. but then leaving `Any` is fine